core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -150,8 +154,7 @@ private static void deleteFiles(FileIO io, Set<ManifestFile> allManifests) {
                 String type = reader.isDeleteManifestReader() ? "delete" : "data";
                 deleteFiles(io, pathsToDelete, type, false);
               } catch (IOException e) {
-                throw new RuntimeIOException(
-                    e, "Failed to read manifest file: %s", manifest.path());
+                LOG.warn("Failed to read manifest file: {}", manifest.path(), e);
```

Comment:
This might be both reading or deleting files, should we update the message?

Reply:
Only [line](https://github.com/apache/iceberg/pull/9184/files#diff-eb6315ae857b4f8efb2c65b93e931432bba5a33b407ed60cf3b8c76f0277cda1R142) can cause this IOException, which is only while reading. All the deleteFile operations are catching related exception and handling it without throwing it back. 

IMO, here it should be error for read only. WDYT ?

Reply:
this part of the code isn't actually being tested, so please add a separate test where this code path would be reached

Reply:
`IOException` can only be thrown by [CloseableGroup](https://github.com/apache/iceberg/blob/36140b819c60743b39176e28a63ab588df445329/api/src/main/java/org/apache/iceberg/io/CloseableGroup.java#L75). I’m struggling to find a better way to mock this call. Do you have any suggestions?

Also, the current error message doesn’t seem very clear to me. It should be something like:` Unexpected behavior: failed to close the resource.`

My intention was to ensure the operation doesn’t fail while deleting files. However, if this exception is encountered, it might be safer to throw an RTE  rather than catching it. What do you think?”